sdk/keyvault/keyvault-certificates/review/keyvault-certificates.api.md

```diff
@@ -76,7 +76,7 @@ export class CertificateClient {
     restoreCertificateBackup(backup: Uint8Array, options?: RestoreCertificateBackupOptions): Promise<KeyVaultCertificateWithPolicy>;
     setContacts(contacts: CertificateContact[], options?: SetContactsOptions): Promise<CertificateContact[] | undefined>;
     updateCertificatePolicy(certificateName: string, policy: CertificatePolicy, options?: UpdateCertificatePolicyOptions): Promise<CertificatePolicy>;
-    updateCertificateProperties(certificateName: string, version: string, options?: UpdateCertificateOptions): Promise<KeyVaultCertificate>;
+    updateCertificateProperties(certificateName: string, version?: string, options?: UpdateCertificateOptions): Promise<KeyVaultCertificate>;
```

Comment:
I believe we generally dont want to have multiple optional parameters.
Either `version` should move into `UpdateCertificateOptions` or we should have an overloads

cc @bterlson to confirm

Reply:
We have the same problem on keys and secrets. For example:

```ts
updateKeyProperties(name: string, keyVersion: string, options?: UpdateKeyPropertiesOptions): Promise<KeyVaultKey>;
```

Reply:
Should we still do this change?

Reply:
I made this issue to do this change separately: https://github.com/Azure/azure-sdk-for-js/issues/6543 since we need to do the change on the three libraries, perhaps this can wait.

Reply:
@sadasant are you making that `keyVersion` optional to? Otherwise it doesn't have the same problem.

However, I agree with Ramya: overload if it's very different usage that should be documented separately, otherwise it's yet another optional thing you can provide.

Reply:
If we are using #6543 to track moving of version to options, then I'd prefer we dont make the change to version being optional parameter in this PR and keep the design in sync between keys, certs and secrets.

Reply:
yes, #6543 would also make keyVersion optional. Please correct me if it is not the same problem as here. But I do appreciate the conclusion that Ramya presented 😁💙

Reply:
To clarify:
- Either update this PR to revert the change being done to make the version optional and handle the requirement as part of #6543
- Or update this PR to move the version into `UpdateCertificateOptions`, use #6543 to do the same for keys & secrets
- Or update this PR to add an overload to `updateCertificateProperties` such that you have the below and use #6543 to do the same for keys & secrets
     - `updateCertificateProperties(certificateName: string, options?: UpdateCertificateOptions)`
     - `updateCertificateProperties(certificateName: string, version: string, options?: UpdateCertificateOptions)`